books/users.go

```diff
@@ -1 +1,111 @@
 package books
+
+import (
+	"encoding/json"
+	"fmt"
+
+	zoho "github.com/schmorrison/Zoho"
+)
+
+// GetCurrentUser will return the currently authenticated users
+// https://www.zoho.com/books/api/v3/users/#get-current-user
+func (c *API) GetCurrentUser() (data CurrentUserResponse, err error) {
+	endpoint := zoho.Endpoint{
+		Name:         "users",
+		URL:          fmt.Sprintf("https://books.zoho.%s/api/v3/users/me", c.ZohoTLD),
+		Method:       zoho.HTTPGet,
+		ResponseData: &CurrentUserResponse{},
+	}
+
+	err = c.Zoho.HTTPRequest(&endpoint)
+	if err != nil {
+		return CurrentUserResponse{}, fmt.Errorf("Failed to retrieve current user: %s", err)
+	}
+
+	if v, ok := endpoint.ResponseData.(*CurrentUserResponse); ok {
+		return *v, nil
+	}
+
+	return CurrentUserResponse{}, fmt.Errorf("Data retrieved was not 'UsersResponse'")
+}
+
+func (m *MorePermissions) UnmarshalJSON(data []byte) error {
+	if string(data) == `""` {
+		return nil
+	}
+
+	type tmp MorePermissions
+	return json.Unmarshal(data, (*tmp)(m))
+}
```

Comment:
I love how predictable Zoho is lol, its an array, unless its empty.

Curious if you need a JSONMarshal method? Are you certain Zoho doesn't expect an empty ```[ ]``` when receiving a request for create or update? Obviously you can adapt when you get there, but may as well consider it now.

Reply:
When creating/updating a user, the role is passed as `role_id` (https://www.zoho.com/books/api/v3/users/#create-a-user). So I don't believe you can set any permission fields directly on the user.

I was actually surprised by how much data the response contained for a single user. Based on the docs I had expected the API to be a lot less chatty. So I would propose to leave it as is.

And yeah, giving us an empty string instead of an empty array was a bit of a curveball. :D